test/rendering_js-strict.test.js

```diff
@@ -0,0 +1,148 @@
+              "doc": "The opacity of a line"
+          },
+          "stroke-linejoin": {
+              "css": "line-join",
+              "default-value": "miter",
+              "type": [
+                  "miter",
+                  "miter-revert",
+                  "round",
+                  "bevel"
+              ],
+              "expression": true,
+              "doc": "The behavior of lines when joining.",
+              "default-meaning": "The line joins will be rendered using a miter look."
+          },
+          "stroke-linecap": {
+              "css": "line-cap",
+              "default-value": "butt",
+              "type": [
+                  "butt",
+                  "round",
+                  "square"
+              ],
+              "expression": true,
+              "doc": "The display of line endings.",
+              "default-meaning": "The line endings will be rendered using a butt look."
+          },
+          "comp-op": {
+              "css": "line-comp-op",
+              "default-value": "overlay",
+              "default-meaning": "Add the current symbolizer on top of other symbolizer.",
+              "doc": "Composite operation. This defines how this symbolizer should behave relative to symbolizers atop or below it.",
+              "type": [
+                  "multiply",
+                  "add",
+                  "overlay"
+              ],
+              "expression": true
+          },
+          "stroke-dasharray": {
+              "css": "line-dasharray",
+              "type": "numbers",
+              "expression": true,
+              "doc": "A pair of length values [a,b], where (a) is the dash length and (b) is the gap length respectively. More than two values are supported for more complex patterns.",
+              "default-value": "none",
+              "default-meaning": "The line will be drawn without dashes."
+          }
+      }
+    }
+  };
+
+  var expectedErrorMessageRegex = /Unrecognized rule: polygon-fill/;
+
+  before(function() {
+    this.referenceData = tree.Reference.data;
+  });
+
+  after(function() {
+    if (this.referenceData) {
+      tree.Reference.setData(this.referenceData);
+    }
+  });
+
+  it('should fail if a feature is not supported and strict is turned on', function () {
+    assert.throws(
+      function () {
+        var RendererJS = new carto.RendererJS({reference: reference, mapnik_version: '1.0.0', strict: true });
+        var shader = RendererJS.render(style);
+      },
+      expectedErrorMessageRegex
+    );
+  });
+
+  function rendererStrictModeOffTest(RendererJS) {
+    return function () {
+      var shader = RendererJS.render(style);
+
+      assert.ok(shader.layers);
+      assert.equal(shader.layers.length, 2);
+    };
+  }
+
+  it('should pass if a feature is not supported but strict mode is not specified', rendererStrictModeOffTest(
+    new carto.RendererJS({reference: reference, mapnik_version: '1.0.0' })
```

Comment:
(IMO) We should assert that it doesn't throw an error. `assert.doesNotThrow` would be  👌